src/lib/containers/discussion_forum/DiscussionThread.tsx

```diff
@@ -124,15 +126,41 @@ export function DiscussionThread(props: DiscussionThreadProps) {
           </Tooltip>
         )}
       </div>
-      {/* Todo : add markdown editor to add replies */}
-      <FormControl type="text" placeholder="Write a reply..." />
+      {!showReplyEditor1 ? (
+        <FormControl
+          type="text"
+          placeholder="Write a reply..."
+          onClick={() => setShowReplyEditor1(true)}
+        />
+      ) : (
+        <ForumThreadEditor
+          id={threadId}
+          isReply={true}
+          onClose={() => setShowReplyEditor1(false)}
+        />
+      )}
```

Comment:
Do you need this here and `showReplyEditor1`? It seems like the other one will always work if you remove `replies.length > 0 &&`

Reply:
I was imitating as much as Synapse was doing. It originally has a textarea which renders the editor onClick which is what the `showReplyEditor` is doing. There was also another separate textarea at the bottom when there were existing replies in the thread.